enigma/src/main/java/org/quiltmc/enigma/api/Enigma.java

```diff
@@ -62,6 +63,13 @@ public class Enigma {
 	public static final String URL = "https://quiltmc.org";
 	public static final int ASM_VERSION = Opcodes.ASM9;
 
+	// These are compile-time constants.
```

Comment:
why are these constant instead of being based on the existing `version.properties`? i can see that it's a deliberate choice, but i'm not clear on the logic behind it

Reply:
I might be derping, but iiuc:
- compile time constants will get inlined, so plugins using the default `supportsEngimaVersion` will "remember" the version they compiled against
- the `VERSION` string is not a compile time constant, so if the default `supportsEngimaVersion` used it, it would always match the run time version, making it pointless

Reply:
Actually I think I'll update that to use `Integer::toString` on the ints instead of `Integer::parseInt` on the strings; a little stricter.